src/submission/schema-migration/migration-manager.ts

```diff
@@ -83,54 +90,49 @@ const submitMigration = async (
 
 const startMigration = async (roMigration: DeepReadonly<DictionaryMigration>) => {
   const migration = _.cloneDeep(roMigration) as DictionaryMigration;
+
   if (!migration._id) {
     throw new Error('Migration should have an id');
   }
+
   const migrationId = migration._id;
   const newSchemaVersion = migration.toVersion;
-  const dryRun = migration.dryRun;
-
-  // analyze changes
-  if (!migration.analysis) {
-    const analysis = await manager
-      .instance()
-      .analyzeChanges(migration.fromVersion, migration.toVersion);
 
-    migration.analysis = analysis;
-    await migrationRepo.update(migration);
-  }
+  const newTargetSchema = await manager
+    .instance()
+    .loadSchemaByVersion(manager.instance().getCurrent().name, newSchemaVersion);
 
-  // get the change analysis from the migration object
-  const changeAnalysis = migration.analysis as ChangeAnalysis;
-
-  // check for breaking changes
-  const invalidatingFields: any = findInvalidatingChangesFields(changeAnalysis);
-
-  await checkDonorDocuments(migration, migrationId, dryRun, invalidatingFields, newSchemaVersion);
+  await checkDonorDocuments(migration, newTargetSchema);
 
   // close migration
-  migration.state = 'CLOSED';
-  migration.stage = 'COMPLETED';
-  await migrationRepo.update(migration);
-  return migration;
+  const updatedMigration = await migrationRepo.getById(migrationId);
+  const migrationToClose = _.cloneDeep(updatedMigration) as DictionaryMigration;
+  if (!migrationToClose) {
+    throw new Error('where did the migration go? expected migration not found');
+  }
+  migrationToClose.state = 'CLOSED';
+  migrationToClose.stage = 'COMPLETED';
+  const closedMigration = await migrationRepo.update(migrationToClose);
+  return closedMigration;
 };
 
 // start iterating over paged donor documents records (that weren't checked before)
-// TODO enhance to report progress of migration to db
-// TODO report when done to slack
 const checkDonorDocuments = async (
   migration: DictionaryMigration,
-  migrationId: string,
-  dryRun: boolean,
-  invalidatingFields: any,
-  newSchemaVersion: string,
+  newSchema: SchemasDictionary,
 ) => {
   let migrationDone = false;
+  if (!migration._id) {
+    throw new Error('Migration should have an id');
+  }
+  const migrationId = migration._id;
+  const dryRun = migration.dryRun;
+
   while (!migrationDone) {
     let invalidCount = 0;
     let validCount = 0;
     const donors = await getNextUncheckedDonorDocumentsBatch(migrationId, 20);
-
+    const newSchemaVersion = newSchema.version;
```

Comment:
could probably set this once outside the loop, no...

Reply:
will remove